diff --git a/lib/client/modules/blog/components/ArticleList.js b/lib/client/modules/blog/components/ArticleList.js
index 14c22eeba8c97abaa9062863f7a7e99a1debe6cc..bb73f01da1d3cbbbc73858b705c41f074263be20 100644
--- a/lib/client/modules/blog/components/ArticleList.js
+++ b/lib/client/modules/blog/components/ArticleList.js
@@ -53,6 +53,13 @@ export default defineComponent({
                     window.scrollTo(0, distance);
                 }, 100);
             });
+            // FIXME: Workaround for https://github.com/vuepress/vuepress-next/issues/1249
+            watch(
+                () => route.query,
+                ({ page }) => {
+                updatePage(page ? Number(page) : 1);
+                }
+            );
         });
         return () => h("div", { id: "article-list", class: "article-wrapper" }, currentArticles.value.length
             ? [