diff --git a/lib/client/components/ArticleHeader.vue b/lib/client/components/ArticleHeader.vue
index 475de8942dd5323f4b1e30dde64749ee673bdb63..8ef0c2746c70c73af26b5680ebc758124cd52353 100644
--- a/lib/client/components/ArticleHeader.vue
+++ b/lib/client/components/ArticleHeader.vue
@@ -1,24 +1,12 @@
 <template>
-  <div
-    class="article-header"
-    :style="headerStyle()"
-    :class="{ 'use-image': frontmatter.useHeaderImage }"
-  >
-    <div
-      v-if="frontmatter.useHeaderImage && frontmatter.headerMask"
-      class="article-header-mask"
-      :style="{ background: frontmatter.headerMask }"
-    />
+  <div class="article-header" :style="headerStyle()" :class="{ 'use-image': frontmatter.useHeaderImage }">
+    <div v-if="frontmatter.useHeaderImage && frontmatter.headerMask" class="article-header-mask"
+      :style="{ background: frontmatter.headerMask }" />
 
     <div class="article-header-content">
       <div v-if="frontmatter.tags" class="article-tags">
-        <RouterLink
-          v-for="(item, index) in frontmatter.tags"
-          :key="index"
-          class="article-tag"
-          :class="{ active: currentTag == item }"
-          :to="getPathByTag(item)"
-        >
+        <RouterLink v-for="(item, index) in frontmatter.tags" :key="index" class="article-tag"
+          :class="{ active: currentTag == item }" :to="getPathByTag(item)">
           {{ item }}
         </RouterLink>
       </div>
@@ -32,10 +20,7 @@
       </p>
 
       <div class="article-icons">
-        <div
-          v-if="frontmatter.author || personalInfo.name"
-          class="article-icon"
-        >
+        <div v-if="frontmatter.author || personalInfo.name" class="article-icon">
           <VIcon name="fa-regular-user" />
           <span>{{ frontmatter.author || personalInfo.name }}</span>
         </div>
@@ -51,17 +36,10 @@
         </div>
       </div>
     </div>
-    <div
-      v-if="frontmatter.useHeaderImage && frontmatter.headerImageCredit"
-      class="article-image-credit"
-    >
+    <div v-if="frontmatter.useHeaderImage && frontmatter.headerImageCredit" class="article-image-credit">
       {{ themeLocale.headerImageCredit }}
-      <a
-        v-if="frontmatter.headerImageCreditLink"
-        :href="frontmatter.headerImageCreditLink"
-        target="_blank"
-        rel="noopener noreferrer"
-      >
+      <a v-if="frontmatter.headerImageCreditLink" :href="frontmatter.headerImageCreditLink" target="_blank"
+        rel="noopener noreferrer">
         {{ frontmatter.headerImageCredit }}
       </a>
       <span v-else>{{ frontmatter.headerImageCredit }}</span>
@@ -70,7 +48,7 @@
 </template>
 
 <script setup lang="ts">
-import { usePageData, usePageFrontmatter, withBase } from "@vuepress/client";
+import { usePageData, usePageFrontmatter } from "@vuepress/client";
 import type {
   GungnirThemePageData,
   GungnirThemePostFrontmatter,
@@ -103,10 +81,26 @@ const headerStyle = () => {
     frontmatter.value.useHeaderImage &&
     frontmatter.value.headerImage
   ) {
-    style.backgroundImage = `url(${withBase(frontmatter.value.headerImage)})`;
+    style.backgroundImage = `url(${triggerUri(frontmatter.value.headerImage)})`;
   }
   return style;
 };
 
+let lastIndex = 0;
+
+function triggerUri(url: string) {
+  if (((url || "").split("?")).length >= 2) {
+    const cdn = (url || "").split("?")[0]
+    let index: string;
+    index = (url || "").split("?")[1].split("=")[1];
+    const max = Math.floor(Math.random() * parseInt(index, 10) + 1);
+    if (lastIndex == 0) {
+      lastIndex = max
+    }
+    return cdn + "/" + lastIndex + ".jpg";
+  }
+  return url;
+}
+
 const personalInfo = themeLocale.value.personalInfo as PersonalConfig;
 </script>
diff --git a/lib/client/components/PostListItem.vue b/lib/client/components/PostListItem.vue
index f04a8e45b48f171405be374abca3e11b3f37798c..c1e3ec4a24ceed63efd7aba13b26bda10cbb2f8d 100644
--- a/lib/client/components/PostListItem.vue
+++ b/lib/client/components/PostListItem.vue
@@ -1,7 +1,7 @@
 <template>
   <div class="post-item">
     <div class="post-item__img" @click="$router.push(item.path)">
-      <img :src="withBase(item.info.headerImage)" />
+      <img :src="triggerUri(item.info.headerImage)" />
     </div>
     <div class="else">
       <p class="post-item__date">
@@ -24,7 +24,6 @@
 </template>
 
 <script setup lang="ts">
-import { withBase } from "@vuepress/client";
 import type { PropType } from "vue";
 import type { GungnirThemePostData } from "../../shared";
 import { formateDateString } from "../utils";
@@ -36,6 +35,22 @@ defineProps({
   }
 });
 
+let lastIndex = 0;
+
+function triggerUri(url: string) {
+  if (((url || "").split("?")).length >= 2) {
+    const cdn = (url || "").split("?")[0]
+    let index: string;
+    index = (url || "").split("?")[1].split("=")[1];
+    const max = Math.floor(Math.random() * parseInt(index, 10) + 1);
+    if (lastIndex == 0) {
+      lastIndex = max
+    }
+    return cdn + "/" + lastIndex + ".jpg";
+  }
+  return url;
+}
+
 const getExcerpt = (excerpt: string) => {
   return excerpt
     .replace(/<RouterLink to/g, "<a href")